### ISSUE TITLE
Exception from NuGet.exe -list (invalid version string)

### ISSUE THREAD

**Issue by scovetta** [NuGet/Home#4951]
The current version of nuget.exe fails to list packages from nuget.org that don't conform to valid semantic versioning:

```
PS C:\test> nuget
NuGet Version: 3.5.0.1996
```

I'm now listing packages for the pre-release package `Cocotte`:

```
PS C:\test> nuget list -Prerelease Cocotte
'1.0.0-201701090444-prerelease' is not a valid version string.
Parameter name: version
```

This also appears when listing all packages (the operation causes nuget.exe to quit when it gets to 'Cocotte').

This also affects the NuGet SDK (similar to bug #3826).

It appears that the server-side and client-side validation don't match, causing these problems.

**Comment by emgarten** [comment]
@scovetta this should be improved in NuGet.exe 4.0.0, would you try it out and let me know if you are still able to hit this issue? You can find it here: https://dist.nuget.org/win-x86-commandline/v4.0.0/nuget.exe

**Comment by scovetta** [comment]
Thanks @emgarten, correct, this does work in NuGet.exe 4.0.0 (tried under Mono):

```
scovetta@prestos:/tmp$ mono ./nuget.exe list -Prerelease Cocotte
Cocotte 1.0.0-201701090444-prerelease
```

**Comment by emgarten** [comment]
This has been fixed with SemVer 2.0.0 support
